src/main/java/cat/udl/eps/softarch/academicrecruit/domain/ProcessStage.java

```diff
@@ -30,4 +33,12 @@
 
     private Date beginDate;
     private Date endDate;
+
+    @ManyToOne
+    @JsonIdentityReference(alwaysAsId = true)
+    private SelectionProcess selectionProcess;
+
+    @OneToOne(mappedBy = "activeProcessStage")
+    @JsonIdentityReference(alwaysAsId = true)
+    private SelectionProcess selectionProcessBeingActive;
```

Comment:
Remember we said we would only set the relation on the dominating side of the relation. You should remove the relation from this side.